### PR TITLE
✨ RW-13: Add plants pictures

### PR DESCRIPTION
**Pull request by kasia-antos**
- Added plants and API for creating and listing plants 
- Changed API `/api/species` to `/api/assigned-species` to be more specific
- Updated `TrackableModel` to have protected `created_by` and `updated_by` fields

### CODE REVIEW

File: backend/pyproject.toml [hunk 1]
```diff
@@ -21,14 +20,16 @@ django-debug-toolbar = "^4.2.0"
...
+pillow = "^10.2.0"
 
 [tool.poetry.group.dev.dependencies]
 flake8 = "^7.0.0"
 black = "^23.12.1"
 isort = "^5.13.2"
-mypy = "^1.7.0"
-django-stubs = {extras = ["compatible-mypy"], version = "^4.2.7"}
-djangorestframework-stubs = {extras = ["compatible-mypy"], version = "^3.14.5"}
+mypy = "^1.6.0"
```
Comment (tulustul) [inline backend/pyproject.toml:30]: Jest jakiś konkretny powód do downgrade'u?

File: backend/replant/api/plant.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            raise serializers.ValidationError(
+                {
+                    "assigned_species_id": [
+                        "Assigned species with provided ID does not exist."
+                    ]
+                }
+            )
+        return assigned_species
+
+    def validate(self, attrs):
```
Comment (tulustul) [inline backend/replant/api/plant.py:49]: `attrs: dict`

File: backend/replant/api/plant.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        assigned_species_id = attrs["assigned_species_id"]
+        user = cast(User, self.context["request"].user)
+
+        assigned_species = self._validate_assigned_species(assigned_species_id, user)
+
+        attrs["assigned_species"] = assigned_species
+        attrs["user"] = user
+        return attrs
+
+    def create(self, validated_data):
```
Comment (tulustul) [inline backend/replant/api/plant.py:59]: `validated_data: dict`

File: backend/replant/models/plant.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+from .utils import TrackableModel
+
+
+class Plant(TrackableModel):
+    class ReviewState(models.TextChoices):
+        PENDING = auto()
+        APPROVED = auto()
+        REJECTED = auto()
+
+    def image_upload_to(self, filename: str) -> str:
```
Comment (tulustul) [inline backend/replant/models/plant.py:17]: Can we put methods below fields?

**Review by tulustul — APPROVED** [review]

File: backend/pyproject.toml (hunk 1 above)
Comment (kasia-antos) [reply backend/pyproject.toml:30]: Well I was getting a lot of `Couldn't resolve related manager for relation..` errors and had no idea what was wrong but after downgrading `django-stubs` from `4.2.7` to `4.2.6` it works fine. I had to downgrade mypy as well to be compatible. I can try to investigate it deeper when I have some free time later.

File: backend/replant/models/plant.py (hunk 4 above)
Comment (kasia-antos) [reply backend/replant/models/plant.py:17]: This one is used by the field it has to be defined before
https://github.com/ulamlabs/replant/blob/RW-13-add-pictures/backend/replant/models/plant.py#L22

**Pull request merged**